engine/runtime/src/main/java/org/enso/interpreter/runtime/error/DataflowError.java

```diff
@@ -110,11 +112,11 @@ private DataflowError(Object payload, Node location) {
     this.ctx = EnsoContext.get(location);
   }
 
-  private DataflowError(Object payload, AbstractTruffleException prototype) {
+  private DataflowError(EnsoContext ctx, Object payload, AbstractTruffleException prototype) {
     super(prototype);
     this.payload = payload;
     this.ownTrace = false;
-    this.ctx = prototype instanceof PanicException panic ? panic.ctx() : EnsoContext.get(null);
```

Comment:
- this _fallback call_ to `EnsoContext.get(null)`
- was yielding the `AssertionError` when `-ea` was on
- the assert is there since #7493
- the functionality wasn't affected, but such a call couldn't be on _fast path_ - hence the `assert`
- passing the real `ctx` around (as done by this PR) seems safer